probe-rs/src/probe/stlink/mod.rs

```diff
@@ -786,27 +779,25 @@ impl<D: StLinkUsb> STLink<D> {
     fn read_mem_32bit(
         &mut self,
         address: u32,
-        length: u16,
+        data: &mut [u8],
         apsel: u8,
-    ) -> Result<Vec<u32>, DebugProbeError> {
+    ) -> Result<(), DebugProbeError> {
         log::debug!(
             "Read mem 32 bit, address={:08x}, length={}",
             address,
-            length
+            data.len()
         );
         // Maximum supported read length is 2^16 bytes.
```

Comment:
Maybe update the byte count in this comment too :)
This number is very odd, isn't it?
```suggestion
        // Maximum supported read length is 6144 bytes.
```